internal/controllers/parser.go

```diff
@@ -102,6 +102,27 @@ func UpdateConfigFromAPIOpts(
 	// store proxied services in map to de-duplicate
 	proxiedServices := map[string]*validation.Service{}
 
+	if opts.Auth != nil && opts.Auth.JWT != nil {
+		paths := []string{}
+		for path := range spec.Paths {
+			paths = append(paths, path)
+		}
+
+		args := &auth.ParseAuthArguments{
+			Logger:                       ctrl.Log,
+			EnvoyConfiguration:           envoyConfiguration,
+			HTTPConnectionManagerBuilder: httpConnectionManagerBuilder,
+			CloudEntityBuilder:           nil,
+			CloudEntityBuilderArguments:  nil,
+			GenerateClusterName:          generateClusterName, // each cluster can be uniquely identified by dns name + port (i.e. canonical Host, which is hostname:port)
+			KubernetesClient:             kubernetesClient,
+		}
+		err := auth.ParseJWTOptions(opts.Auth.JWT, args, paths)
```

Comment:
Can't you use spec.Paths instead of creating another slice of paths?
```suggestion
		err := auth.ParseJWTOptions(opts.Auth.JWT, args, spec.Paths)
```
and then in ParseJWTOptions get values you need?

Reply:
I don't see the need to pass in the rather large `spec` object when all that is required is the list of `paths`.

Are you OK if I just leave this as-is?

Reply:
Jasmin isn't saying pass the whole spec, just spec.Paths. What's the difference to what you're doing?

Reply:
It's a big object:

```
// Paths is specified by OpenAPI/Swagger standard version 3.
// See https://github.com/OAI/OpenAPI-Specification/blob/main/versions/3.0.3.md#paths-object
type Paths map[string]*PathItem
```

The only thing I need is the keys in there so there's no point in passing in the whole object to only then just retrieve the keys, i.e., it makes no use of `PathItem`.

Please mark as resolved, if you're happy with it.

Reply:
resolved